code/modules/organs/external/damage.dm

```diff
@@ -91,19 +91,28 @@
 
 			switch(damage_type)
 				if(BRUTE)
-					if(edge_eligible && (amount + prev_brute) >= max_damage * DROPLIMB_THRESHOLD_EDGE)
-						droplimb(TRUE, DROPLIMB_EDGE)
-					else if((amount + prev_brute) >= max_damage * DROPLIMB_THRESHOLD_DESTROY)
-						droplimb(FALSE, DROPLIMB_BLUNT)
-					else if((amount + prev_brute) >= max_damage * DROPLIMB_THRESHOLD_TEAROFF)
-						droplimb(FALSE, DROPLIMB_EDGE)
-					else if(brute_dam && BP_IS_ROBOTIC(src))
-						droplimb(prob(50), pick(DROPLIMB_EDGE, DROPLIMB_BLUNT))
+					if(status & ORGAN_BROKEN)
+						if(edge_eligible && (amount + prev_brute) >= max_damage * DROPLIMB_THRESHOLD_EDGE)
+							droplimb(TRUE, DROPLIMB_EDGE)
+						else if((amount + prev_brute) >= max_damage * DROPLIMB_THRESHOLD_DESTROY)
+							droplimb(FALSE, DROPLIMB_BLUNT)
+						else if((amount + prev_brute) >= max_damage * DROPLIMB_THRESHOLD_TEAROFF)
+							droplimb(FALSE, DROPLIMB_EDGE)
+						else if(BP_IS_ROBOTIC(src) && edge_eligible && (amount + prev_brute) >= max_damage * DROPLIMB_THRESHOLD_EDGE)
+							droplimb(TRUE, DROPLIMB_EDGE)
+						else if(BP_IS_ROBOTIC(src) && (amount + prev_brute) >= max_damage * DROPLIMB_THRESHOLD_DESTROY)
+							droplimb(FALSE, DROPLIMB_BLUNT)
+						else if(BP_IS_ROBOTIC(src) && (amount + prev_brute) >= max_damage * DROPLIMB_THRESHOLD_TEAROFF)
+							droplimb(FALSE, DROPLIMB_EDGE)
```

Comment:
What do the extra `BP_IS_ROBOTIC(src)` checks do, shouldn't it be already covered by the earlier checks?

Reply:
Done. They are indeed were unnessesery. Commented those if anybody in the future will deside to create separate conditions for FBP's